File: src/client/classes/api/req_res/update-timeis-request.ts
'use strict'

import { TimeIs } from '@/classes/datas/time-is'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateTimeisRequest extends GkillAPIRequest {

    timeis: TimeIs

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.timeis = new TimeIs()
        this.want_response_kyou = false
    }

}

File: src/server/gkill/api/req_res/update_timeis_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateTimeisRequest struct {
	SessionID string `json:"session_id"`

	TimeIs reps.TimeIs `json:"timeis"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}
